cli/flox/src/commands/environment.rs

```diff
@@ -1362,20 +1368,79 @@ impl Pull {
 
         let pointer_content =
             serde_json::to_string_pretty(&pointer).context("Could not serialize pointer")?;
-        let pointer_path = dot_flox_path.join(ENVIRONMENT_POINTER_FILENAME);
 
-        fs::create_dir_all(&dot_flox_path).context("Could not create .flox/ directory")?;
+        let dot_flox_parent = dot_flox_path
+            .parent()
+            .context("Could not get .flox/ parent")?;
+        fs::create_dir_all(dot_flox_parent).context("Could not create .flox/ parent directory")?;
+
+        let temp_dot_flox_dir = tempfile::TempDir::with_prefix_in(DOT_FLOX, dot_flox_parent)
+            .context("Could not create temporary directory for cloning environment")?;
+
+        let pointer_path = temp_dot_flox_dir.path().join(ENVIRONMENT_POINTER_FILENAME);
         fs::write(pointer_path, pointer_content).context("Could not write pointer")?;
 
-        let result =
-            ManagedEnvironment::open(flox, pointer, &dot_flox_path).map_err(Self::convert_error);
+        let mut env = {
+            let result = Dialog {
+                message,
+                help_message: None,
+                typed: Spinner::new(|| ManagedEnvironment::open(flox, pointer, &temp_dot_flox_dir)),
+            }
+            .spin()
+            .map_err(Self::convert_error);
+
+            match result {
+                Err(err) => {
+                    fs::remove_dir_all(&temp_dot_flox_dir)
+                        .context("Could not clean up .flox/ directory")?;
+                    Err(err)?
+                },
+                Ok(env) => env,
+            }
+        };
+
+        let result = Dialog {
+            message,
+            help_message: None,
+            typed: Spinner::new(|| env.build(flox)),
+        }
+        .spin();
+
         match result {
-            Err(err) => {
-                fs::remove_dir_all(dot_flox_path).context("Could not clean up .flox/ directory")?;
-                Err(err)?;
+            Ok(_) => {
+                fs::rename(temp_dot_flox_dir, dot_flox_path)
+                    .context("Could not move .flox/ directory")?;
+            },
+            Err(
+                e @ EnvironmentError2::Core(CoreEnvironmentError::LockedManifest(
+                    LockedManifestError::BuildEnv(CallPkgDbError::PkgDbError(PkgDbError {
+                        exit_code: 100,
+                        ..
+                    })),
+                )),
+            ) => {
+                if !add_systems && !query_add_system(&flox.system) {
+                    bail!(formatdoc! {"
+                        This environment is not yet compatible with your system ({system}).
```

Comment:
suggestion blocking: don't print this twice:
```
> The environment you are trying to pull is not compatible with your system (aarch64-darwin). Don't pull this environment
ERROR: This environment is not yet compatible with your system (aarch64-darwin).
```

Reply:
I'll error out with "Pull aborted" now (instead of printing the error again) if we prompted and the user chose to abort.
In a way that's a success case to me but i guess that's splitting hairs

Reply:
Yes, I would want to add it after "Did not pull the environment." I will make a separate ticket to update this message when we have the link to the correct doc to put here. So we can call this done.